### PR TITLE
GH Actions: selectively use `fail-fast` with setup-php

### PR DESCRIPTION
**Pull request by jrfnl**
## Description
I've seen some recent build failures due to the `setup-php` action running into a rate limit and not downloading the required version of Composer. In the case of this action, that would make the test runs worthless.

The `setup-php` action runner defaults to _showing_ these type errors in the logs, but not stopping the workflow run.

So, specifically for those jobs where the Composer version is important, I'm adding the `fail-fast` option to `setup-php` to fail the build if the action runner ran into any errors.

Ref: https://github.com/shivammathur/setup-php#fail-fast-optional

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] I have added tests to cover my changes.

---

### :new: GH Actions: update PHP version for `composer-normalize`

Looks like the `composer-normalize` package has dropped support for PHP < 8.0 as of version `2.29.0`.

Refs:
* https://github.com/ergebnis/composer-normalize/releases/tag/2.29.0
* https://github.com/ergebnis/composer-normalize/pull/998

### CODE REVIEW

**Comment by jrfnl** [comment]
I've added a second commit to fix the build which was failing due to an unrelated change.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@ jobs:
           php-version: "latest"
           tools: "composer:${{ matrix.composer-version }}"
           coverage: "none"
+          fail-fast: true
```
Comment (jrfnl) [inline .github/workflows/continuous-integration.yml:51]: Just realized I implemented this incorrectly, it should be:
```suggestion
        env:
          fail-fast: true
```

Fixed in an updated commit.

**Review by ramsey — APPROVED** [review]

**Comment by danielbachhuber** [comment]
> GH Actions: update PHP version for composer-normalize
> Looks like the composer-normalize package has dropped support for PHP < 8.0 as of version 2.29.0.

@jrfnl Would this break any downstream use of the `ramsey/composer-install` action?

Our deployments stopped working ~2 weeks ago:

image
image

https://github.com/wp-cli/wp-cli-bundle/actions/workflows/deployment.yml
https://github.com/wp-cli/wp-cli-bundle/actions/runs/3898207401/jobs/6656726095#step:13:49

@schlessera

**Comment by jrfnl** [comment]
> > GH Actions: update PHP version for composer-normalize
> > Looks like the composer-normalize package has dropped support for PHP < 8.0 as of version 2.29.0.
> 
> @jrfnl Would this break any downstream use of the `ramsey/composer-install` action?

Absolutely not. This change only affects the CI of the `ramsey/composer-install` action and has no effect on the actual functionality.

Without looking into your issue, but just based on a quick glance at error shown in the screenshots you included, I have a niggly feeling your issue may have something to do with Composer 2.x vs 2.2 LTS.

Composer 2.3 dropped support for PHP < 7.2 and the autoload files generated by Composer 2.3+ _should_ still be compatible with PHP 5.6 - 7.1, but no guarantee. As Composer 2.5 and 2.5.1 came out about three weeks ago, there may be something in those which broke the autoload compatibility with PHP < 7.2 ?
https://github.com/composer/composer/releases

Composer 2.2 is a LTS release, so can be used if you need to be sure autoload files are PHP cross version compatible.

**Comment by danielbachhuber** [comment]
> Without looking into your issue, but just based on a quick glance at error shown in the screenshots you included, I have a niggly feeling your issue may have something to do with Composer 2.x vs 2.2 LTS.

@jrfnl Looks like you are correct! https://github.com/composer/composer/issues/11245#issuecomment-1382413537

> Absolutely not. This change only affects the CI of the `ramsey/composer-install` action and has no effect on the actual functionality.

Thanks for being kind to my ignorance 😊 I'm still sorting through how all of these GitHub Actions are chained together...

**Comment by jrfnl** [comment]
@danielbachhuber No worries, hope you can sort it out and get your own CI running again soon.

**Comment by jrfnl** [comment]
@danielbachhuber FYI: I just saw that Composer 2.5.2 has been released which contains a fix for the PHP 5.6 autoloader snafu.

**Pull request merged**